aiida/orm/implementation/sqlalchemy/utils.py

```diff
@@ -80,11 +80,12 @@ def __setattr__(self, key, value):
             self._flush(fields=fields)
 
     def is_saved(self):
-        """Retun whether the wrapped model instance is saved in the database.
+        """Return whether the wrapped model instance is saved in the database.
 
         :return: boolean, True if the model is saved in the database, False otherwise
         """
-        return self._model.id is not None
+        with self._model.session.no_autoflush:
```

Comment:
```suggestion
        # we should not flush here since it may lead to IntegrityErrors
        # which are handled later in the save method
        with self._model.session.no_autoflush:
```